lib/rules/key-spacing.js

```diff
@@ -356,10 +500,16 @@ module.exports = {
                 widths = properties.map(getKeyWidth), // Width of keys, including quotes
                 targetWidth = Math.max.apply(null, widths),
                 i, property, whitespace, width,
-                align = multiLineOptions.align,
+                align = alignmentOptions.on,
                 beforeColon = multiLineOptions.beforeColon,
                 afterColon = multiLineOptions.afterColon,
-                mode = multiLineOptions.mode;
+                mode = alignmentOptions.mode;
+
+            if (alignmentOptions && length > 1) { // When aligning values within a group, use the alignment configuration.
+                beforeColon = alignmentOptions.beforeColon;
+                afterColon = alignmentOptions.afterColon;
+                mode = alignmentOptions.mode;
+            }
```

Comment:
Rather than initializing these variables on declaration and then potentially overriding them here, I'd rather declare them without initializing them and then set them in one branch of an `if`/`else` here. I think that's slightly easier to reason about.